src/cascadia/UIHelpers/TextMenuFlyout.cpp

```diff
@@ -0,0 +1,201 @@
+#include "pch.h"
+#include "TextMenuFlyout.h"
+
+#include <LibraryResources.h>
+
+#include "TextMenuFlyout.g.cpp"
+
+using namespace ::winrt::Windows::UI::Xaml;
+using namespace ::winrt::Windows::UI::Xaml::Controls;
+using namespace ::winrt::Windows::ApplicationModel::Resources::Core;
+using namespace ::winrt::Microsoft::UI::Xaml::Controls;
+using namespace ::winrt::Windows::System;
+using namespace ::winrt::Windows::UI::Xaml::Input;
+
+using MenuFlyoutItemClick = void (*)(IInspectable const&, RoutedEventArgs const&);
+
+constexpr auto NullSymbol = static_cast<Symbol>(0);
+
+namespace winrt::Microsoft::Terminal::UI::implementation
+{
+#pragma warning(suppress : 26455) // Default constructor should not throw. Declare it 'noexcept' (f.6).
+    TextMenuFlyout::TextMenuFlyout()
+    {
+        // Most of the initialization is delayed until the first call to MenuFlyout_Opening.
+        Opening({ this, &TextMenuFlyout::MenuFlyout_Opening });
+    }
+
+    void TextMenuFlyout::MenuFlyout_Opening(IInspectable const&, IInspectable const&)
+    {
+        auto target = Target();
+        if (!target)
+        {
+            return;
+        }
+        hstring selection;
+        bool writable = false;
+
+        // > Common group of selectable controls with common actions
+        // > The I in MIDL stands for...
+        // No common interface.
+        if (const auto box = target.try_as<NumberBox>())
```

Comment:
What about ComboBox with `IsEditable=True`?

Reply:
It seems like we don't use those yet. I wonder if it's worth adding support for that in advance...